react-exercises/modal-react/src/App.js

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import Modal from './modal/modal';
+import Backdrop from './backdrop/backdrop';
+import './App.css';
+
+function App() {
+  
+  const [isLoginOpen, setIsLoginOpen] = useState(false);
+  const [isSignupOpen, setIsSignupOpen] = useState(false);
+  const [isMyBioOpen, setIsMyBioOpen] = useState(false);
+
+  useEffect(() => {
+    document.onkeydown = onKeyDown;
+  },[onkeydown])
+
+  const close = () => {
+    setIsSignupOpen(false); 
+    setIsLoginOpen(false);
+    setIsMyBioOpen(false);
+  }
+
+  const onKeyDown = (event) => {
+    let modal = document.querySelector('.modal');
+    let focusableElementsString = `a[href], area[href], input:not([disabled]), button:not([disabled]), 
+                                    iframe, object, embed, [tabindex="0"], [contendeditable]`;
+
+    let focusableElements = (modal !== null) ? modal.querySelectorAll(focusableElementsString) :
+                                               document.querySelectorAll(focusableElementsString);
+    focusableElements = Array.prototype.slice.call(focusableElements);
+    let firstElement = focusableElements[0];
+    let lastElement = focusableElements[focusableElements.length - 1];
+
+    if(event.keyCode === 9) {
+        if(event.shiftKey){
+          if(document.activeElement === firstElement ){
+              event.preventDefault();
+              lastElement.focus();
+          }
+        } else {
+          if(document.activeElement === lastElement ){
+            event.preventDefault();
+            firstElement.focus();
+          }
+        }
+    }
+
+    if(event.keyCode === 27 && modal !== null) {
+      close();
+    }
```

Comment:
This should be a part of the Modal component

Reply:
Correct @ahhda ......focus logic should be part of the modal....working on those changes now.